File: src/test/java/SortCodeSubstitutionTests.java
import com.dambra.paul.moduluschecker.Account.BankAccount;
import com.dambra.paul.moduluschecker.SortCodeSubstitution;
import com.google.common.collect.ImmutableMap;
import org.junit.Test;

import java.io.IOException;

import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.CoreMatchers.theInstance;
import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.core.IsEqual.equalTo;
import static org.hamcrest.core.IsNot.not;

public class SortCodeSubstitutionTests {
    @Test
    public void SortCodesNotInTheTableAreUnchanged() {

        ImmutableMap<String,String> substitutions = ImmutableMap.<String, String>builder()
                                                .put("not_the_sort_code", "12345")
                                                .build();

        SortCodeSubstitution sortCodeSubstitution = new SortCodeSubstitution(substitutions);
        BankAccount originalAccount = BankAccount.Of("012345", "01234567");
        BankAccount account = sortCodeSubstitution.Apply(originalAccount);

        assertThat(account.sortCode, is(equalTo("012345")));
        assertThat(account.accountNumber, is(equalTo("01234567")));
        assertThat(account, is(not(theInstance(originalAccount))));
    }

    @Test
    public void SortCodesInTheTableAreChanged() {

        ImmutableMap<String,String> substitutions = ImmutableMap.<String, String>builder()
                .put("012345", "543210")
                .build();

        SortCodeSubstitution sortCodeSubstitution = new SortCodeSubstitution(substitutions);
        BankAccount originalAccount = BankAccount.Of("012345", "01234567");
        BankAccount account = sortCodeSubstitution.Apply(originalAccount);

        assertThat(account.sortCode, is(equalTo("543210")));
        assertThat(account.accountNumber, is(equalTo("01234567")));
        assertThat(account, is(not(theInstance(originalAccount))));
    }

    @Test
    public void CanLoadFromFileResource() throws IOException {
        SortCodeSubstitution sortCodeSubstitution = SortCodeSubstitution.fromFile("file/scsubtab.txt");

        BankAccount ba = BankAccount.Of("938173", "01234567");
        assertThat(sortCodeSubstitution.Apply(ba).sortCode, is(equalTo("938017")));
    }
}

File: src/test/java/SortCodeSubstitutionTests.kt


import com.github.pauldambra.moduluschecker.SortCodeSubstitution
import com.github.pauldambra.moduluschecker.account.BankAccount
import com.natpryce.hamkrest.assertion.assert
import com.natpryce.hamkrest.equalTo
import com.natpryce.hamkrest.sameInstance
import org.junit.Test
import java.io.IOException

class SortCodeSubstitutionTests {
    @Test
    fun sortCodesNotInTheTableAreUnchanged() {

        val substitutions = mapOf("not_the_sort_code" to "12345")

        val sortCodeSubstitution = SortCodeSubstitution(substitutions)
        val originalAccount = BankAccount("012345", "01234567")
        val account = sortCodeSubstitution.apply(originalAccount)

        assert.that(account.sortCode, equalTo("012345"))
        assert.that(account.accountNumber, equalTo("01234567"))
        assert.that(account, !sameInstance(originalAccount))
    }

    @Test
    fun sortCodesInTheTableAreChanged() {

        val substitutions = mapOf("012345" to "543210")

        val sortCodeSubstitution = SortCodeSubstitution(substitutions)
        val originalAccount = BankAccount("012345", "01234567")
        val account = sortCodeSubstitution.apply(originalAccount)

        assert.that(account.sortCode, equalTo("543210"))
        assert.that(account.accountNumber, equalTo("01234567"))
        assert.that(account, !sameInstance(originalAccount))
    }

    @Test
    @Throws(IOException::class)
    fun canLoadFromFileResource() {
        val sortCodeSubstitution = SortCodeSubstitution.fromFile("file/scsubtab.txt")

        val ba = BankAccount("938173", "01234567")
        assert.that(sortCodeSubstitution.apply(ba).sortCode, equalTo("938017"))
    }
}
